main/autogen/autoloader.cc

```diff
@@ -400,17 +407,84 @@ void AutoloadWriter::write(const core::GlobalState &gs, const AutoloaderConfig &
                            UnorderedSet<std::string> &toDelete, const DefTree &node) {
     string name = node.root() ? "root" : node.name().show(gs);
     string filePath = join(path, fmt::format("{}.rb", name));
-    FileOps::writeIfDifferent(filePath, node.renderAutoloadSrc(gs, alCfg));
+    if (!alCfg.packagedAutoloader || !node.root()) {
+        FileOps::writeIfDifferent(filePath, node.renderAutoloadSrc(gs, alCfg));
```

Comment:
Are these files marked `typed: ignore`? If not, can we do that so that no one ever accidentally tries to typecheck them (since now they will fail in packaging mode)?

Reply:
I think that depends on the header we supply with `--autogen-autoloader-preamble`. I'm nervous about hard-coding it here, in case it conflicts with later preamble decisions, but it's easy to change the preamble at use site.